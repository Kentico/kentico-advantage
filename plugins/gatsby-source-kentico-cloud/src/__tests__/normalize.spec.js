
const normalize = require('../normalize');
const contentType = require('./contentType.json');
const contentTypeNodes = require('./contentTypeNodes.json');
const contentItem = require('./contentItem.json');
// const contentItemNodes = require('./contentItemNodes');

describe('createContentTypeNode with correct arguments', () => {
  it(`creates a content type node`, () => {
    const createNodeId = jest.fn();
    createNodeId.mockReturnValue(`aea6da0c-4130-593c-8b6e-006e6bace1de`);

    expect(
        normalize.createContentTypeNode(createNodeId, contentType)
    ).toMatchSnapshot();
  });
});

describe('createContentItemNode with correct arguments', () => {
  it(`creates a content item node`, () => {
    const createNodeId = jest.fn();
    createNodeId.mockReturnValue(`362bd0da-5b1a-533b-9575-107c2e3c6931`);

    expect(
        normalize.createContentItemNode(
            createNodeId, contentItem, contentTypeNodes
        )
    ).toMatchSnapshot();
  });
});
