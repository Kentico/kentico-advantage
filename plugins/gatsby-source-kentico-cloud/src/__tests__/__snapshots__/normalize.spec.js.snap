// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createContentItemNode with correct arguments creates a content item node 1`] = `
Object {
  "children": Array [],
  "contentType___NODE": "aea6da0c-4130-593c-8b6e-006e6bace1de",
  "elements": Object {},
  "id": "362bd0da-5b1a-533b-9575-107c2e3c6931",
  "internal": Object {
    "content": "{\\"system\\":{\\"id\\":\\"7e70cb89-9281-4db9-8130-36076ee1e698\\",\\"name\\":\\"Project reference 1\\",\\"codename\\":\\"project_reference_1\\",\\"language\\":\\"default\\",\\"type\\":\\"project_reference\\",\\"sitemap_locations\\":[],\\"last_modified\\":\\"2018-05-29T13:40:17.3224207Z\\"},\\"elements\\":{}}",
    "contentDigest": "bf808dfcd7f72d75b3f2dc427d0587dd",
    "type": "KenticoCloudItemProjectReference",
  },
  "otherLanguages___NODE": Array [],
  "parent": null,
  "system": Object {
    "codename": "project_reference_1",
    "id": "7e70cb89-9281-4db9-8130-36076ee1e698",
    "language": "default",
    "last_modified": "2018-05-29T13:40:17.3224207Z",
    "name": "Project reference 1",
    "sitemap_locations": Array [],
    "type": "project_reference",
  },
  "usedByContentItems___NODE": Array [],
}
`;

exports[`createContentTypeNode with correct arguments creates a content type node 1`] = `
Object {
  "children": Array [],
  "contentItems___NODE": Array [],
  "elements": Object {
    "finished_at": Object {
      "name": "Finished at",
      "type": "date_time",
    },
    "name___teaser_image__name": Object {
      "name": "Name",
      "type": "text",
    },
    "name___teaser_image__teaser_image": Object {
      "name": "Teaser image",
      "type": "asset",
    },
    "started_at": Object {
      "name": "Started at",
      "type": "date_time",
    },
    "url": Object {
      "name": "URL",
      "type": "text",
    },
    "url_slug": Object {
      "name": "URL slug",
      "type": "url_slug",
    },
  },
  "id": "aea6da0c-4130-593c-8b6e-006e6bace1de",
  "internal": Object {
    "content": "{\\"system\\":{\\"id\\":\\"9ea98d20-a477-455c-b30f-0e3b9d055795\\",\\"name\\":\\"Project reference\\",\\"codename\\":\\"project_reference\\",\\"last_modified\\":\\"2018-05-30T12:39:13.115595Z\\"},\\"elements\\":{\\"url\\":{\\"type\\":\\"text\\",\\"name\\":\\"URL\\"},\\"name___teaser_image__teaser_image\\":{\\"type\\":\\"asset\\",\\"name\\":\\"Teaser image\\"},\\"started_at\\":{\\"type\\":\\"date_time\\",\\"name\\":\\"Started at\\"},\\"name___teaser_image__name\\":{\\"type\\":\\"text\\",\\"name\\":\\"Name\\"},\\"finished_at\\":{\\"type\\":\\"date_time\\",\\"name\\":\\"Finished at\\"},\\"url_slug\\":{\\"type\\":\\"url_slug\\",\\"name\\":\\"URL slug\\"}}}",
    "contentDigest": "b9f5d773962088bb97a6cde3e4f803ce",
    "type": "KenticoCloudTypeProjectReference",
  },
  "parent": null,
  "system": Object {
    "codename": "project_reference",
    "id": "9ea98d20-a477-455c-b30f-0e3b9d055795",
    "last_modified": "2018-05-30T12:39:13.115595Z",
    "name": "Project reference",
  },
  "usedByContentItems___NODE": Array [],
}
`;
